### DISCUSSION TITLE
SwaggerUI form for 'fastapi.security.OAuth2AuthorizationCodeBearer' is using old/cached AuthCode when logging-out/logging-in

### DISCUSSION THREAD

**Discussion by vladislav-krastev** [fastapi/fastapi#13012] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.security import OAuth2AuthorizationCodeBearer
dep = OAuth2AuthorizationCodeBearer(
    authorizationUrl="https://accounts.google.com/o/oauth2/v2/auth",
    tokenUrl=f"http://localhost:8082/auth/tester-token",
    scopes={
        "https://www.googleapis.com/auth/userinfo.profile": "allow profile",
        "https://www.googleapis.com/auth/userinfo.email": "allow email",
    }
)
MyDependency = Annotated[str, Depends(dep)]


@router_auth.post("/tester-token")
async def tester_token(request: Request):
    from fastapi.datastructures import QueryParams
    body_parsed = QueryParams(await request.body())
    print(body_parsed["code"])
    ... # use 'body_parsed["code"]' to manually fetch the actual AccessToken from Google ...
```

### Description

Hello, everyone!

_DISCLAIMER:_ I realize the issue is almost 100% caused by a bug (or intentional design choice I don't understand?) in the SwaggerUI, but I want to make sure our community haven't figured out a way around it already. The relevant (same) question I posted on their thread: [10229](https://github.com/swagger-api/swagger-ui/discussions/10229)
Will make sure to update and close both threads when their is an answer in either one.

Logging-in and Logging-out is done via the big green 'Authorize' button.

The `authorizationUrl` param of the SwaggerUI form gets called only on the first logging-in and only that time a new AuthCode is fetched. Any subsequent logging-out and logging-in reuses the same AuthCode obtained the first time.

Any ideas how to force a call to the `authorizationUrl` every time someone logs-in?

Thank you in advance!

### Operating System

Other

### Operating System Details

WSL on WIndows 10

### FastAPI Version

0.115.5

### Pydantic Version

2.10.2

### Python Version

3.12.7

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, please create an MRE
